pipeline/README.md

```diff
@@ -0,0 +1,136 @@
+├── test_extract.py      # Unit tests for the `extract.py` module
+├── test_transform.py    # Unit tests for the `transform.py` module
+├── test_load.py         # Unit tests for the `load.py` module
+└── README.md            # Documentation for the project
+```
+
+---
+
+## Modules
+
+### **1. `extract.py`**
+- **Purpose:** Fetches earthquake data from the USGS API.
+- **Key Functionality:**
+  - Queries the API and retrieves earthquake data for the last minute.
+  - Extracts attributes such as magnitude, location, and type.
+  - Handles errors, including missing data and API connection issues.
+- **API URL Used:** [USGS Earthquake Feed](https://earthquake.usgs.gov/earthquakes/feed/v1.0/summary/all_hour.geojson)
+- **Tests:** `test_extract.py`
+
+### **2. `transform.py`**
+- **Purpose:** Cleans and validates the data extracted by `extract.py`.
+- **Key Functionality:**
+  - Cleans and standardizes fields like `felt`, `cdi`, and `alert`.
+  - Validates geographic (latitude/longitude) and magnitude ranges.
+  - Drops invalid records and prepares data for database insertion.
+- **Tests:** `test_transform.py`
+
+### **3. `load.py`**
+- **Purpose:** Inserts cleaned data into a PostgreSQL database.
+- **Key Functionality:**
+  - Connects to the database using environment variables for credentials.
+  - Resolves foreign keys for related tables (e.g., alert types, magnitude types).
+  - Batch-inserts earthquake records into the `earthquakes` table.
+- **Tests:** `test_load.py`
+
+### **4. `etl.py`**
+- **Purpose:** Runs the entire ETL pipeline.
+- **Key Functionality:**
+  - Combines extraction, transformation, and loading into a single script.
+  - Can be run as a standalone process or triggered by an AWS Lambda event.
+  - Provides logging and error handling for end-to-end execution.
+
+### **5. `schema.sql`**
+- **Purpose:** Defines the PostgreSQL database schema.
+- **Key Functionality:**
+  - Creates tables for earthquakes, alerts, magnitude types, and other entities.
+  - Includes constraints for data integrity (e.g., latitude/longitude ranges).
+
+---
+
+## Testing
+
+### **1. `test_extract.py`**
+- **Purpose:** Ensures data extraction is correct.
+- **Tests Include:**
+  - Validation of extracting the correct data for the correct time.
+  - Correct handling of missing fields and empty requests.
+  - Verification of the correct output format.
+
+### **2. `test_transform.py`**
+- **Purpose:** Ensures data transformations are correct.
+- **Tests Include:**
+  - Validation of latitude, longitude, magnitude, and CDI values.
+  - Correct handling of missing or invalid fields.
+  - Verification of the cleaned output format.
+
+### **3. `test_load.py`**
+- **Purpose:** Validates data insertion logic into the database.
+- **Tests Include:**
+  - Correct mapping of foreign keys.
+  - Successful batch inserts into the `earthquakes` table.
+  - Handling of invalid data during database operations.
```

Comment:
Would be nice to add a test coverage score, for now just a line and near the end we can put the score in.